gapis/trace/android/adreno/profiling_data.go

```diff
@@ -90,6 +90,45 @@ func extractTraceHandles(ctx context.Context, replayHandles *[]int64, replayHand
 	}
 }
 
+func fixContextIds(contextIDs []int64) []int64 {
+	// This is a temporary work to workaround a QC bug
+	// that causes first deviceID to be zero after a
+	// renderpass change in the same queue submit.
+
+	zeroIndices := make([]int, 0)
```

Comment:
Using a separate list of indices here looks like it may be more complicated than necessary?

Did you consider just doing two full passes:
1. Detect if there are multiple non-zero ids.
2. If there is only one non-zero id, just do a full second pass.

Depending on the size of the input, this may not even be less efficient. (Re)allocating the zeroIndices also has some cost. (No idea if performance is a concern here.)

Reply:
Performance is not a concern here. It's all under 100 entry and all of them are probably in cache anyway.

And it does pretty much the same thing. I don't see a reason to change.

Reply:
I considered to mention the same as Matthias, since I am pretty sure that two full passes are still cheaper than the array allocation(s), but then decided not to mention it because I agree that performance is not a concern here.

Reply:
I was thinking explicitly marking the zero indices and fixing them later, actually explains what code is doing better than just passing on entire array in the second run but I am open to suggestions.

Because people in the future may ask, why do we run through the entire array and set all elements to the deviceID which is already there. This way, I think it's more clear and explains itself that, we specifically search for zeros, and if there is only one deviceID, we fill them with that handle.